From 75ccc43c9e1702c99abe89832e592e6f8d328612 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 04 Nov 2020 14:13:22 +0200
Subject: [PATCH] [SW-12755] gaudi: security: split pcie rr to rd/rw

As a preparation for securing dynamically the versions and FIT images,
split the PCIe RR secure function to separate read a and write.
This will enable opening only for read access (e.g. versions).

Change-Id: Ia63fc888c545219d27148d1d73ec56896c9ba79d
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index ad851da..388f168 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -1046,7 +1046,7 @@
 	 */
 }
 
-static void preboot_secure_pcie_rr(int rr, u64 min64, u64 max64)
+static void preboot_secure_pcie_rr_wr(int rr, u64 min64, u64 max64)
 {
 	u32 min = (u32)min64;
 	u32 max = (u32)max64;
@@ -1057,6 +1057,14 @@
 	regmap_writel(min, rr_dest);
 	rr_dest = rr_base + mmPCIE_WRAP_RR_WR_PRV_RANGE_REG_MAX_0;
 	regmap_writel(max, rr_dest);
+}
+
+static void preboot_secure_pcie_rr_rd(int rr, u64 min64, u64 max64)
+{
+	u32 min = (u32)min64;
+	u32 max = (u32)max64;
+	u32 rr_base = rr * sizeof(u32), rr_dest;
+
 	/* Set RP RR */
 	rr_dest = rr_base + mmPCIE_WRAP_RR_RD_PRV_RANGE_REG_MIN_0;
 	regmap_writel(min, rr_dest);
@@ -1064,6 +1072,12 @@
 	regmap_writel(max, rr_dest);
 }
 
+static void preboot_secure_pcie_rr(int rr, u64 min64, u64 max64)
+{
+	preboot_secure_pcie_rr_wr(rr, min64, max64);
+	preboot_secure_pcie_rr_rd(rr, min64, max64);
+}
+
 static void preboot_secure_lbw_rr(int rr, u64 min64, u64 max64)
 {
 	int i;
